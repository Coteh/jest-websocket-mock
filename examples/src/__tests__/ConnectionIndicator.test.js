import React from "react";
import renderer from "react-test-renderer";
import makeStore from "../store";
import { actions } from "../store/reducer";
import ConnectionIndicator from "../ConnectionIndicator";

describe("The ConnectionIndicator component", () => {
  it("renders a green dot when successfully connected", () => {
    const store = makeStore();
    store.dispatch(actions.connectionSuccess());
    const component = renderer.create(<ConnectionIndicator store={store} />);
    expect(component).toMatchSnapshot();
  });

  it("renders a red dot when not connected", () => {
    const store = makeStore();
    store.dispatch(actions.connectionLost());
    const component = renderer.create(<ConnectionIndicator store={store} />);
    expect(component).toMatchSnapshot();
  });
});
