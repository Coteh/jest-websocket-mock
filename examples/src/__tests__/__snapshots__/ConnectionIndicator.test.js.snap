// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The ConnectionIndicator component renders a green dot when successfully connected 1`] = `
<div
  className="ConnectionIndicator ConnectionIndicator--connected"
/>
`;

exports[`The ConnectionIndicator component renders a red dot when not connected 1`] = `
<div
  className="ConnectionIndicator ConnectionIndicator--disconnected"
/>
`;
